File: src/main/java/me/nobaboy/nobaaddons/features/chatcommands/impl/party/AllInviteCommand.java
package me.nobaboy.nobaaddons.features.chatcommands.impl.party;

import me.nobaboy.nobaaddons.NobaAddons;
import me.nobaboy.nobaaddons.features.chatcommands.ChatContext;
import me.nobaboy.nobaaddons.features.chatcommands.IChatCommand;
import me.nobaboy.nobaaddons.util.ChatUtils;
import me.nobaboy.nobaaddons.util.PartyUtils;

import java.util.Collections;
import java.util.List;

public class AllInviteCommand implements IChatCommand {
    @Override
    public String name() {
        return "allinvite";
    }

    @Override
    public String usage() {
        return "(allinvite|allinv)";
    }

    @Override
    public void run(ChatContext ctx) {
        if(!PartyUtils.isLeader) return;
        ChatUtils.delayedSend("p settings allinvite");
    }

    @Override
    public boolean isEnabled() {
        return NobaAddons.config.partyAllInviteCommand;
    }

    @Override
    public List<String> aliases() {
        return Collections.singletonList("allinv");
    }
}

File: src/main/kotlin/me/nobaboy/nobaaddons/features/chatcommands/impl/party/AllInviteCommand.kt
package me.nobaboy.nobaaddons.features.chatcommands.impl.party

import me.nobaboy.nobaaddons.NobaAddons
import me.nobaboy.nobaaddons.api.PartyAPI
import me.nobaboy.nobaaddons.features.chatcommands.ChatContext
import me.nobaboy.nobaaddons.features.chatcommands.IChatCommand
import me.nobaboy.nobaaddons.util.HypixelCommands

class AllInviteCommand : IChatCommand {
    override val name: String = "allinvite"

    override val aliases: MutableList<String> = mutableListOf("allinv")

    override val usage: String = "(allinvite|allinv)"

    override val isEnabled: Boolean
        get() = NobaAddons.config.chatCommands.partyCommands.allInvite

    override fun run(ctx: ChatContext) {
        if (PartyAPI.isLeader()) return
        HypixelCommands.partyAllInvite()
    }
}